src/coreclr/jit/ssabuilder.cpp

```diff
@@ -141,9 +141,19 @@ void SsaBuilder::ComputeDominanceFrontiers(BasicBlock** postOrder, int count, Bl
         // Otherwise, there are > 1 preds.  Each is a candidate B2 in the definition --
         // *unless* it dominates "block"/B3.
 
+        FlowGraphDfsTree*       dfsTree = m_pCompiler->m_dfsTree;
+        FlowGraphDominatorTree* domTree = m_pCompiler->m_domTree;
+
         for (FlowEdge* pred = blockPreds; pred != nullptr; pred = pred->getNextPredEdge())
         {
-            DBG_SSA_JITDUMP("   Considering predecessor " FMT_BB ".\n", pred->getSourceBlock()->bbNum);
+            BasicBlock* predBlock = pred->getSourceBlock();
+            DBG_SSA_JITDUMP("   Considering predecessor " FMT_BB ".\n", predBlock->bbNum);
+
+            if (!dfsTree->Contains(predBlock))
+            {
+                DBG_SSA_JITDUMP("    Unreachable node\n");
```

Comment:
This was a preexisting bug. The below loop would read the `bbIDom` set by the old dominator computation in this case.